### DISCUSSION TITLE
Example patterns written by secure erase or secure erase enhanced for various drives

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#269] [General]
### Seagate ST3500312CS Video 3.5, Firmware version SC13, 500GB

I've started this discussion to document the sort of data that is finally written to a drive when the drives own firmware is responsible for the erase.

This information will be useful to learn more about the data that can be expected and if repeatable and not random, how I can then write code that verifies a secure erase.

Starting with a Seagate ST3500312CS Video 3.5 500GB HDD, Firmware version SC13 - Erased using **Secure Erase Enhanced** which writes a repeating pattern 0x33CC55AA on this particular model and firmware.

![Screenshot_20240731_150308](https://github.com/user-attachments/assets/ded662e6-bb8b-4580-8c91-5f7ebcedb7c4)
------------------
![Screenshot_20240731_165246](https://github.com/user-attachments/assets/b3a1ef17-e11f-480a-933c-e470ff9ed4ba)
Same drive as above but **Secure Erase** selected, which writes zeros, as opposed to **Secure Erase Enhanced** which writes a repeating pattern.

**Comment by PartialVolume** [comment]
### Western Digital WDC WD20EURS-63SPKY0, firmware revision 80.00A80, 2TB

For both `secure erase` and `security erase enhanced` the resulting pattern is all zeros. 278 minutes to complete security erase. 278 minutes to complete security erase enhanced. Makes you wonder what a enhanced security erase does that's different if the time is the same and the pattern is the same.
![Screenshot_20240801_123552](https://github.com/user-attachments/assets/70b3a914-2446-4429-87ab-dd86d53400e3)

**Comment by PartialVolume** [comment]
### Seagate EXOS ST000NM0007-2A1101, firmware revision SN02, 12TB

Secure erase enhanced writes 0x33CC55AA

![Screenshot_20240801_145423](https://github.com/user-attachments/assets/3b420b86-c37d-405c-aa0b-66627ba58acc)

**Comment by NeOTecH01** [comment]
> ### Seagate ST3500312CS Video 3.5, Firmware version SC13, 500GB
> 
> I've started this discussion to document the sort of data that is finally written to a drive when the drives own firmware is responsible for the erase.
> 
> This information will be useful to learn more about the data that can be expected and if repeatable and not random, how I can then write code that verifies a secure erase.
> 
> Starting with a Seagate ST3500312CS Video 3.5 500GB HDD, Firmware version SC13 - Erased using **Secure Erase Enhanced** which writes a repeating pattern 0x33CC55AA on this particular model and firmware.
> ## ![Screenshot_20240731_150308](https://private-user-images.githubusercontent.com/22084881/353865143-ded662e6-bb8b-4580-8c91-5f7ebcedb7c4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODgxMTY2NTMsIm5iZiI6MTc4ODExNjM1MywicGF0aCI6Ii8yMjA4NDg4MS8zNTM4NjUxNDMtZGVkNjYyZTYtYmI4Yi00NTgwLThjOTEtNWY3ZWJjZWRiN2M0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODMwVDE4NTkxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2ZDNmMTRiYzU4MjRkNTUzYTRiMTdmMjI2MWI5Mzk4M2QwNThmNWU3MzA1N2U5OTVmNzY3YjRkYzUwZWE3NTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.o10lDJKKkMtM5CvurmuTJdnptsww86D2JChk5wJe3jE)
> 
> ![Screenshot_20240731_165246](https://private-user-images.githubusercontent.com/22084881/353900551-b3a1ef17-e11f-480a-933c-e470ff9ed4ba.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODgxMTY2NTMsIm5iZiI6MTc4ODExNjM1MywicGF0aCI6Ii8yMjA4NDg4MS8zNTM5MDA1NTEtYjNhMWVmMTctZTExZi00ODBhLTkzM2MtZTQ3MGZmOWVkNGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwODMwVDE4NTkxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0MGYxOTBiZmQwY2ExN2Y3OTliZGM4ZTBkM2ViN2FhMjU1NzE0NmJkM2FkYmU2NmJkY2ZiMjBlZWZmM2M2OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.6ef5nzC8SpwKfnXIySE1b35eW_WExTF5P_g9PnHling) Same drive as above but **Secure Erase** selected, which writes zeros, as opposed to **Secure Erase Enhanced** which writes a repeating pattern.

can u please provide me with Firmware  for Seagate ST3500312CS 3.5, Firmware version SC13, 500GB?

**Reply by PartialVolume** [reply]
>can u please provide me with Firmware for Seagate ST3500312CS 3.5, Firmware version SC13, 500GB?

You want to know the firmware version on the drive I was using?

**Comment by NeOTecH01** [comment]
if u got a backup firmware of  Seagate Pipeline HD 500GB ST3500312CSc ver
sc13, i have an issue with it im not recovering data just want it to work
again, thank you very much

**Reply by PartialVolume** [reply]
>if u got a backup firmware of  Seagate Pipeline HD 500GB ST3500312CSc ver sc13, i have an issue with it im not recovering data just want it to work
again, thank you very much.

Sorry, I can't help you with that, I'm not even sure Seagate provide that firmware, the drive may even be factory locked so can't be updated. In what way doesn't it work, have you tried wiping it with the latest ShredOS?

**Reply by NeOTecH01** [reply]
thank you so much, the hd spin's well, not recognized by system after cleaning the board it recognized as unknown disk using diskpart it read it as 0 GB, i found the flash software on this website do u think i should i gave it a chance
https://files.hddguru.com/download/SeDiv%20Support/Seagate/Hepburn/

**Reply by PartialVolume** [reply]
Hddguru is a legitimate site so you shouldn't have a problem with malware. However it may depend on what your plans are for the drive. Gemini sums it up here:

Evaluating the safety of a URL involves looking at both the hosting domain and the nature of the files found there.
1. The Domain (hddguru.com)
 * Context: HDDGURU is a well-known, long-standing community and forum specifically focused on hard drive repair, data recovery, and specialized diagnostic utilities.
 * Reputation: It is generally considered a legitimate resource by the professional data recovery community. The files.hddguru.com subdomain acts as a repository for firmware files, manuals, and repair tools shared among technicians.
2. The Content (/download/SeDiv Support/Seagate/Hepburn/)
 * What it is: This path points to support files for SeDiv (a specialized hard drive repair and diagnostic software often used by technicians to work with firmware, clear SMART data, or fix Seagate/Western Digital drives). "Hepburn" refers to a specific Seagate drive family codename.
 * Safety & Security Risks:
   * Not official manufacturer software: These are third-party, low-level diagnostic and repair files. They are not provided or officially endorsed by Seagate.
   * False Positives / Malware Flags: Because tools like SeDiv interact directly with a hard drive's service area and firmware modules (behaving similarly to rootkits or low-level flash tools), antivirus software frequently flags them as potentially unwanted programs (PUPs), trojans, or generic malware.
   * Risk of Drive Damage: Beyond software safety, using low-level tools incorrectly on hard drive firmware can permanently brick or destroy the drive.
Verdict
 * Is it a malicious phishing site or a drive-by malware distributor? No, the domain is a known data recovery community repository.
 * Are the files safe to download and run blindly? Proceed with caution. Antivirus programs will likely flag the contents, and running unverified low-level firmware utilities carries inherent risks to your system and hardware. Only download from here if you are an experienced data recovery technician who understands how to isolate and vet these specific tools.
 *

**Comment by NeOTecH01** [comment]
i have no other choice,the files are clean i'm gonna give it a tray, any
advice regarding the flash process itself? i'm using  rt809h.
